### PR TITLE
image: build and upload as one step

### PR DESCRIPTION
**Pull request by malt3**
### Context

GitHub Actions [has changed the API for Artifacts](https://github.com/actions/toolkit/tree/main/packages/artifact#v2---whats-new). This limits the amount of artifacts per workflow, doesn't allow overwriting artifacts in a Workflow run and changed the way globs are handled.
Since the old API will likely be disabled soon, we have to refactor the image build pipeline.

As a nice bonus, the pipeline is now roughly twice as fast (1h -> 30min).

### Proposed change(s)

- use [uplosi](https://github.com/edgelesssys/uplosi) instead of custom Go code for CSP-specific OS image upload
- add bazel targets under `//image/system:upload_*` for uploading images
- refactor CI workflow to use single job for upload (removed usages of GitHub Artifacts)

### Additional info

- [Build pipeline (console)](https://github.com/edgelesssys/constellation/actions/runs/7421162779)
- [Build pipeline (debug)](https://github.com/edgelesssys/constellation/actions/runs/7420179616)

Debug image for testing:

```
ref/feat-image-build-and-upload-as-one-step/stream/debug/v2.15.0-pre.0.20240105103004-3d9404f299d6
```

Console image for testing:

```
ref/feat-image-build-and-upload-as-one-step/stream/console/v2.15.0-pre.0.20240105121312-66211185bb14
```

Command for uploading debug images from your workstation: `bazel run //image/system:upload_debug -- --ref foo --upload-measurements --fake-sign`

### Checklist

- [x] Add labels (e.g., for changelog category)
- [x] Is PR title adequate for changelog?
- [x] Link to Milestone

### CODE REVIEW

File: image/upload/internal/cmd/measurementsenvelope.go [hunk 1]
```diff
@@ -96,6 +102,94 @@ func runEnvelopeMeasurements(cmd *cobra.Command, _ []string) error {
 	return nil
 }
 
+func overrideStaticMeasurements(in measurements.M, csp cloudprovider.Provider, attestationVariant string) (measurements.M, error) {
```
Comment (daniel-weisse) [inline image/upload/internal/cmd/measurementsenvelope.go:105]: We might want to define the PCRs of this function as const/vars somewhere instead for easier management

File: image/upload/internal/cmd/uplosi.go [hunk 2]
```diff
@@ -0,0 +1,116 @@
+/*
+Copyright (c) Edgeless Systems GmbH
+
+SPDX-License-Identifier: AGPL-3.0-only
+*/
+
+package cmd
+
+import (
+	_ "embed" // go embed requires a blank import
```
Comment (daniel-weisse) [inline image/upload/internal/cmd/uplosi.go:10]: This file does not embed anything, or am I mistaken?

**Review by daniel-weisse — COMMENTED** [review]
Not an in-depth review, mostly skimmed the changes

File: image/upload/internal/cmd/measurementsenvelope.go (hunk 1 above)
Comment (malt3) [reply image/upload/internal/cmd/measurementsenvelope.go]: Moved this as vars in `internal/attestation/measurements/overrides.go`

**Review by msanft — APPROVED** [review]
Went over the new Bazel rule and the Go code invoking / parsing uplosi, which both look fine to me.

**Pull request merged**